Adds openmrs integration guide (#519)
medic/cht-docs: content/en/apps/guides/integrations/openmrs.md

| end group                     |                   |                                    |          |                     |            |            |                     |             |                |      |         |

##### Choices

| list_name         | name | label           |
| ----------------- | ---- | --------------- |
| client_consented            | _1065_Yes_99DCT  | Yes             |
| client_consented            | _1066_No_99DCT   | No              |
| disability_type            | _1058_vision_99DCT   | Vision              |
| disability_type            | _1059_hearing_99DCT   | Hearing              |
| disability_type            | _1060_mental_99DCT   | Mental              |
| disability_type            | _1061_other_99DCT   | Other              |

Remember to convert and upload your forms
```zsh
medic-conf --url=https://<username>:<password>@localhost --accept-self-signed-certs convert-contact-forms upload-contact-forms convert-app-forms upload-app-forms
```

> Remember to setup the [outbound push]({{< ref "apps/reference/app-settings/outbound" >}}) modules to send data to OpenMRS.

Reviewer: You could have this as a note.

#### Handling the data

After collecting data using the forms defined above, the next step is to process and persist it in OpenMRS. Processing includes: 

1. **Cleaning**

> First the data has to be transformed to an OpenMRS-compatible format, before it is queued. This means that you need to define custom RESTful endpoints if not already existing, that would be utilized by the Outbound push modules configured above. Transformation basically involves extracting form data into an object. Most importantly, `discriminators`, which are like flags appended to form data to inform the type of data being processed. For example a `registration` discriminator implies that we're dealing with demographic details. Sample transformed payload is shown below.

Reviewer: ```suggestion
First, the data has to be transformed to an OpenMRS-compatible format before it is queued. This means that you need to define custom RESTful endpoints if not already existing, that would be utilized by the Outbound push modules configured above. Transformation basically involves extracting form data into an object. Most importantly, `discriminators`, which are like flags appended to form data to inform the type of data being processed. For example, a `registration` discriminator implies that we're dealing with demographic details. A sample transformed payload is shown below:
```

You need to define a queue processor that will be consuming the queue and push each object to the rightful handler.

3. **Data handlers**

Data handlers are responsible for persisting the data in OpenMRS. The handler can also be used to trigger feedback to submitters of the data (this can be a brief summary such as number of synced documents and success rate). The following steps apply:

 - Define handlers for each object type (e.g. person / patient, trace report.
 - Create a scheduler to start the queue processor above. The queue processor shall get to the handler for processing.
 - Closely monitor the errors log for prompt action where necessary.

You may want to further configure a service that relays feedback to the CHT. Feel free to utilize in-app text messages, which can be triggered via the [sms endpoint](https://docs.communityhealthtoolkit.org/apps/reference/api/#post-apisms) of CHT. The health workers would receive these feedback messages on their phones as well as access via the Messages tab in-app.

#### Error Handling

> Exceptions thrown during processing can be added to a queue and presented on an interface for action. The erroring data may then be re-queued. You need to consider what works best at this point, but for convenience purposes, it makes sense for a backend user to resolve such errors.

Reviewer: ```suggestion
Exceptions thrown during processing can be added to a queue and presented on an interface for action. The erring data may then be re-queued. You need to consider what works best at this point. For convenience, it makes sense for a backend user to resolve such errors.
```

medic-conf --url=https://<username>:<password>@localhost --accept-self-signed-certs convert-contact-forms upload-contact-forms convert-app-forms upload-app-forms
```

> **Note**: Remember to setup the [Outbound push]({{< ref "apps/reference/app-settings/outbound" >}}) modules to send data to OpenMRS.

Reviewer: ```suggestion
{{% alert title="Note" %}}Remember to setup the [Outbound push]({{< ref "apps/reference/app-settings/outbound" >}}) modules to send data to OpenMRS. {{% /alert %}}
```
Updated this to [blockquote styling](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#notes-and-tips).
